src/Zikula/CoreBundle/Composer/MetaData.php

```diff
@@ -88,7 +89,7 @@ public function __construct(array $json = [])
         $this->dependencies = $this->formatDependencies($json);
         $this->shortName = $json['extra']['zikula']['short-name'];
         $this->class = $json['extra']['zikula']['class'];
-        $this->namespace = mb_substr($this->class, 0, mb_strrpos($this->class, '\\') + 1);
+        $this->namespace = (string) s($this->class)->beforeLast('\\', true);
```

Comment:
space here.

Also, it seems strange to me that a component called **String** requires to be cast as a string... shouldn't it output a string? I know that it is returning an `AbstractSting` that is `\Stringable`, so can we lose the repeated `(string)` casting?

Reply:
The Stringable interface is introduced with PHP 8. Like shown in https://symfony.com/blog/new-in-symfony-5-1-string-improvements#use-stringable-interface the Polyfill provides it already.

> This will allow you to use union types such as string|Stringable in the future when you later upgrade to PHP 8.

At the moment our methods expect `string` for both return types and arguments. So to avoid BC breaks I added the casts in the first step. Changed them to calling `__toString()` now (a3e4402784b5bc3c34cdda98a1859de59690116c).

Reply:
Please add a ticket to 4.0 to change type hints to `\Stringable` where possible?

Reply:
Follow-up created in #4516